spark/src/test/scala/org/apache/spark/sql/delta/DeltaRetentionSuite.scala

```diff
@@ -322,4 +323,171 @@ class DeltaRetentionSuite extends QueryTest
       assert(log.snapshot.numOfSetTransactions == 0)
     }
   }
+
+  protected def cleanUpExpiredLogs(log: DeltaLog): Unit = {
+    val snapshot = log.update()
+
+    val checkpointVersion = snapshot.logSegment.checkpointProvider.version
+    logInfo(s"snapshot version: ${snapshot.version} checkpoint: $checkpointVersion")
+
+    log.cleanUpExpiredLogs(snapshot)
+  }
+
+  for (v2CheckpointFormat <- V2Checkpoint.Format.ALL_AS_STRINGS)
+  test(s"sidecar file cleanup [v2CheckpointFormat: $v2CheckpointFormat]") {
```

Comment:
We should be setting the `v2CheckpointFormat` in the test also - currently it is only used in name.